mlflow/johnsnowlabs.py

```diff
@@ -0,0 +1,874 @@
+        ):
+            return PipelineModel.load(mlflowdbfs_path)
+
+    sparkml_model_uri = append_to_uri_path(model_uri, flavor_conf["model_data"])
+    local_sparkml_model_path = os.path.join(local_mlflow_model_path, flavor_conf["model_data"])
+    return _load_model(
+        model_uri=sparkml_model_uri,
+        dfs_tmpdir_base=dfs_tmpdir,
+        local_model_path=local_sparkml_model_path,
+    )
+
+
+def _load_pyfunc(path, spark=None):
+    """
+    Load PyFunc implementation. Called by ``pyfunc.load_model``.
+    :param path: Local filesystem path to the MLflow Model with the ``johnsnowlabs`` flavor.
+    :param spark: Optionally pass spark context when using pyfunc as UDF. required, because we cannot fetch the Sparkcontext inside of the Workernode which executes the UDF
+    :return:
+    """
+    return _PyFuncModelWrapper(_load_model(model_uri=path), spark if spark else get_or_create_sparksession(path))
+
+
+def get_or_create_sparksession(model_path=None):
+    """
+    1. Check if SparkSession running and get it
+    2. If none exists, create a new one using jars in model_path
+    3. If model_path not defined, rely on nlp.start() to create a new
+    one using johnsnowlabs Jar resolution method
+    See https://nlp.johnsnowlabs.com/docs/en/jsl/start-a-sparksession
+    and https://nlp.johnsnowlabs.com/docs/en/jsl/install_advanced
+    :param model_path:
+    :return:
+    """
+    validate_env_vars()
+    from mlflow.utils._spark_utils import _get_active_spark_session
+    spark = _get_active_spark_session()
+    if spark is None:
+        spark_conf = {}
+        spark_conf["spark.python.worker.reuse"] = "true"
+        # disable task retry (i.e. make it fast fail)
+        spark_conf["spark.task.maxFailures"] = "1"
+        # Disable simplifying traceback from Python UDFs
+        spark_conf["spark.sql.execution.pyspark.udf.simplifiedTraceback.enabled"] = "false"
+        # Show jvm side stack trace.
+        spark_conf["spark.sql.pyspark.jvmStacktrace.enabled"] = "true"
+        spark_conf["spark.jars.excludes"] = "net.sourceforge.f2j:arpack_combined_all"
+        spark_conf["spark.driver.extraJavaOptions"] = "-Dio.netty.tryReflectionSetAccessible=true"
+        spark_conf["spark.executor.extraJavaOptions"] = "-Dio.netty.tryReflectionSetAccessible=true"
+
+        os.environ['PYSPARK_PYTHON'] = sys.executable
+        os.environ['PYSPARK_DRIVER_PYTHON'] = sys.executable
+        if model_path:
+            jar_paths, license_path = fetch_deps_from_path(os.path.join(model_path))
+            # jar_paths += get_mleap_jars().split(',')  # TODO when to load MLleap Jars
+            if license_path:
+                if 'JSL_NLP_LICENSE' not in os.environ:
+                    os.environ.update({k: str(v) for k, v in json.load(open(license_path)).items() if v is not None})
+                    os.environ['JSL_NLP_LICENSE'] = json.load(open(license_path))['HC_LICENSE']
```

Comment:
```suggestion
  with open(license_path) as f:
                    os.environ['JSL_NLP_LICENSE'] = json.load(f)['HC_LICENSE']
```

We should use `with`.

Reply:
Thanks updated